### PR TITLE
Add new features on app

### PR DESCRIPTION
**Pull request by bh-2024**
Here are changes added on react component and also python change

### CODE REVIEW

File: Indie_Gems_Portal/scripts/sort_genres.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# File: process_genres.py
```
Comment (bh-2024) [inline Indie_Gems_Portal/scripts/sort_genres.py:1]: Here are the most important considerations from the code review suggestions:

1. **Add comprehensive testing**: Ensure that the `read_and_sort_genres` function behaves as expected in all scenarios, including edge cases.
2. **Improve error handling**:
	* Catch and handle `FileNotFoundError` when opening the file for reading.
	* Re-attempt to parse the file with a fallback value (e.g., an empty list) if the parsing fails.
3. **Enhance code readability**:
	* Use more descriptive variable names that follow conventional Python naming conventions.
	* Break down long functions into smaller, more focused ones.
	* Add docstrings and comments explaining what the code does, why it's written that way, or how to use it.
4. **Improve type hints**: Include type hints for the `filename` parameter, variables, and return value of `read_and_sort_genres`.
5. **Optimize performance**:
	* Consider reading the file in smaller chunks or using a streaming parser to reduce memory usage.
	* Take advantage of concurrent processing to speed up the parsing process if the file is very large.

These suggestions aim to improve the overall quality, readability, and maintainability of the code while addressing potential issues with functionality, error handling, and performance.

File: Indie_Gems_Portal/scripts/sort_genres_test [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# File: test_sort_genres.py
```
Comment (bh-2024) [inline Indie_Gems_Portal/scripts/sort_genres_test:1]: Here are the most important considerations based on the code review:

1. **Missing imports**: Add `import unittest` to the `setUp` method.
2. **Unused variables**: Remove unused variables from the code.
3. **Testing approach**: Expand test cases to cover different scenarios and edge cases, such as empty data, invalid data, large datasets, and different types of input data.
4. **Test names**: Use a consistent naming convention for test names (e.g., `test_` prefix).
5. **Input handling**: Improve input handling to properly handle different types of input data.

These suggestions will help improve the robustness and maintainability of the test script.

File: Indie_Gems_Portal/src/components/GenreList.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import {
   Spinner,
   Text,
 } from "@chakra-ui/react";
+import React, {useEffect, useState} from "react";
```
Comment (bh-2024) [inline Indie_Gems_Portal/src/components/GenreList.tsx:11]: Based on the provided code review suggestions, here are the most important considerations:

1.  **Proper Documentation**: Add documentation to explain why specific imports are required for this component.

2.  **Initialize Separate State Variable `newData`**: Initialize `newData` with an empty array or object instead of leaving it uninitialized.

3.  **Use of `useEffect`**: Review the dependency list and consider adding logging or debugging statements to understand its behavior.

4.  **Performance Implications**: Review the component's performance and ensure that it does not introduce any unexpected side effects.

5.  **Code Readability**: Use more descriptive variable names, such as `data` instead of `newData`, to improve code readability.

These suggestions aim to address potential issues with code organization, maintainability, and performance while ensuring optimal functionality and usability.
